src/error-table.rkt

```diff
@@ -1,20 +1,365 @@
+                      [(and (infinite? larg-val)
+                            (infinite? rarg-val)
+                            (not (same-sign? larg-val rarg-val))
+                            (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+
+                      ; inf rescue:
+                      ; R(inf) + y = non inf value (inf rescue)
+                      [(and (or (infinite? larg-val)
+                                (infinite? rarg-val))
+                            (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; High condition number:
+                      ; CN(+, x, y) = |x / x + y| 
+                      [(or (> cond-x 1e2) (> cond-y 1e2))
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
+                   
+                   [(list '-.f64 larg rarg)
+                    #:when (or (is-inexact? larg) (is-inexact? rarg))
+                    (define larg-val (hash-ref exacts-hash larg))
+                    (define rarg-val (hash-ref exacts-hash rarg))
+                    (define x-y (- larg-val rarg-val))
+                    (define cond-x (abs (/ larg-val x-y)))
+                    (define cond-y (abs (/ rarg-val x-y)))
+                    
+                    (cond
+                      ; Undeflow sanity check:
+                      ; When x - y correctly underflows, CN is high even though
+                      ; the answer is correct
+                      [(and (= x-y 0.0) (underflow? subexpr)) #f]
+
+                      ; nan rescue:
+                      ; inf - inf = nan but should actually get an inf
+                      [(and (infinite? larg-val)
+                            (infinite? rarg-val)
+                            (same-sign? larg-val rarg-val)
+                            (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+
+                      ; inf rescue
+                      ; If x or y overflow and the other arg rescues it
+                      [(and (or (infinite? larg-val)
+                                (infinite? rarg-val))
+                            (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+
+                      ; High condition number:
+                      ; CN(+, x, y) = |x / x - y|
+                      [(or (> cond-x 1e2) (> cond-y 1e2))
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
+                   
+                   #|
+                   TODO Make this actually work
+                   |#
+                   [(list 'sin.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val  (flabs (hash-ref exacts-hash arg)))
+                    (and (> arg-val 1e30) (mark-erroneous! subexpr pt))]
```

Comment:
I mean, I know where you got this but would be good to be more explicit about the condition number calculation & how it’s broken up. Also, what about, like, sin(pi)? Missing other half of condition. NUmber.

Reply:
I haven't touched the trig operations since we decided not to touch them for while. I think for now I'll just calculate the condition number and check if its greater than 100. I am still unsure how to split, for example sin, xtan(x) into x separately for the large x case and the x being a multiple of pi case. Like, what is the explanation when x is very large multiple of pi?

Reply:
"Cancellation in reduction"?